sdk/nodejs/runtime/rpc.ts

```diff
@@ -30,6 +30,20 @@ export type OutputResolvers = Record<
     (value: any, isStable: boolean, isSecret: boolean, deps?: Resource[], err?: Error) => void
 >;
 
+// Fix to cover cases such as https://github.com/pulumi/pulumi/issues/17736 where an array or other
+// nested object contains an Output, and it's toString method tries to call toString on that output.
+function safeToString(v: any): string {
+    try {
+        return `${v}`;
+    } catch (err) {
+        // We only want to catch our toString errors here. Anything else we should still bubble up to the user.
+        if (err instanceof OutputToStringError) {
```

Comment:
I'm wondering if there's a possible rare edge case when there are multiple SxS copies of `@pulumi/pulumi` at play where the `instanceof` won't work. Wondering if we should protect against that possibility with `OutputToStringError.isInstance`.

Reply:
Good point